pkg/util/net/http.go

```diff
@@ -120,3 +120,28 @@ func GetHTTPClient(req *http.Request) string {
 	}
 	return "unknown"
 }
+
+// Extracts and returns the clients IP from the given request.
+func GetClientIP(req *http.Request) (net.IP, error) {
```

Comment:
If you never return an error, maybe don't have it as a return parameter, just document that this is "best effort" therefore no error, but no promise of accuracy either.